packages/notebook/src/widget.ts

```diff
@@ -553,6 +559,8 @@ export class StaticNotebook extends WindowedList {
     args: sharedModels.NotebookChange
   ): void {
     if (args.cellsChange) {
+      (this.layout as NotebookWindowedLayout).header = null;
```

Comment:
Probably better to call a protected method `clearHeader` in which you dispose the header.